### PR TITLE
Add wrapper for `GDALGetRasterUnitType`

### PR DESCRIPTION
**Pull request by ChristianBeilschmidt**
- [X] I agree to follow the project's [code of conduct](https://github.com/georust/gdal/blob/master/CODE_OF_CONDUCT.md).
- [x] I added an entry to `CHANGES.md` if knowledge of this change could be valuable to users.
---

I've added a wrapper for `GDALGetRasterUnitType` into rasterband. Since I had no test file that has a unit, I added a test file from the original GDAL repo:  https://github.com/OSGeo/gdal/blob/96096ae1e61b1b7afdc686ab9a00ca4fee772ba6/autotest/gdrivers/data/usgsdem/114p01_0100_deme_truncated.dem

### CODE REVIEW

File: src/raster/rasterband.rs [hunk 1]
```diff
@@ -449,6 +449,13 @@ impl<'a> RasterBand<'a> {
             Ok(RasterBand::from_c_rasterband(self.dataset, overview))
         }
     }
+
+    /// Return the unit of the rasterband.
+    /// If there is no unit, the empty string is returned.
```
Comment (lnicola) [inline src/raster/rasterband.rs:454]: I can't find this in the GDAL docs, can we have an `assert_ne!(str_ptr, std::ptr::null())` or something like that?

**Comment by lnicola** [comment]
Wow, I love the undelimited numbers in that `.dem` file.

File: src/raster/rasterband.rs [hunk 2]
```diff
@@ -476,6 +476,18 @@ impl<'a> RasterBand<'a> {
...
+    pub fn unit(&self) -> String {
+        let str_ptr = unsafe { gdal_sys::GDALGetRasterUnitType(self.c_rasterband) };
+
+        if str_ptr.is_null() {
```
Comment (lnicola) [inline src/raster/rasterband.rs:485]: :+1:

File: src/raster/rasterband.rs (hunk 1 above)
Comment (ChristianBeilschmidt) [reply src/raster/rasterband.rs:481]: [f747c81](https://github.com/georust/gdal/pull/271/commits/f747c817b3dc39d0e3803519f9c58052ae961c4c)

**Comment by ChristianBeilschmidt** [comment]
> Wow, I love the undelimited numbers in that `.dem` file.

If you have another file at hand with a unit, I can use it. I just got one from the osgeo gdal repo.

**Comment by lnicola** [comment]
> If you have another file at hand with a unit, I can use it. I just got one from the osgeo gdal repo.

No, it's fine, I was just surprised that GDAL actually parses that.

**Review by lnicola — APPROVED** [review]

**Comment by lnicola** [comment]
I'll wait a bit, then merge this, it's pretty trivial.

**Pull request merged**